### PR TITLE
[Guide]: Adding Netlify CMS to Gatsby

### PR DESCRIPTION
**Pull request by krismorf**
**Summary**
Adding Netlify CMS to Gatsby Guide #2092

### CODE REVIEW

**Comment by erquhart** [comment]
@krismorf here's an issue from someone that walked through the guide: https://github.com/netlify/netlify-cms/issues/2120

**Comment by krismorf** [comment]
@erquhart it was solved in https://github.com/netlify/netlify-cms/pull/2117/commits/a7ea13d4589c94058897c8692091e17b8282006d

**Comment by erquhart** [comment]
Awesome!

**Comment by erquhart** [comment]
@verythorough have time to give this a review?

File: website/content/docs/using-netlify-cms-with-gatsby.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+Far far away, behind the word mountains, far from the countries Vokalia and
+Consonantia, there live the blind texts.
+```
+
+We can see above that each blog post has a title, a date, a description and a body. Now, let's recreate this using Netlify CMS. 
+
+## Add Netlify CMS to your site
+
+First let's install some dependencies. We'll need `netlify-cms` and `gatsby-plugin-netlify-cms`.
```
Comment (verythorough) [inline website/content/docs/using-netlify-cms-with-gatsby.md]: I'd clarify here that the command should be run within the `blog` directory. Or to specify in the "open your text editor" step that one should `cd` into the blog directory and run all future commands in there (regardless of how people actually open their favorite text editor).

File: website/content/docs/using-netlify-cms-with-gatsby.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Using Netlify CMS with Gatsby
```
Comment (verythorough) [inline website/content/docs/using-netlify-cms-with-gatsby.md]: In order for this page to appear on the docs site, you'll need to move this title into a front matter section with `title`, `weight`, and `group`. 

I think `guides` probably makes most sense for the `group` (agree, @tomrutgers @erquhart ?)

For the weight, I'd pick a lower number than [custom-widgets](https://github.com/netlify/netlify-cms/blob/master/website/content/docs/custom-widgets.md), which is 35. Maybe 20? That would leave a fair amount of room on either side.

File: website/content/docs/using-netlify-cms-with-gatsby.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+We can see above that each blog post has a title, a date, a description and a body. Now, let's recreate this using Netlify CMS. 
+
+## Add Netlify CMS to your site
+
+First let's install some dependencies. We'll need `netlify-cms` and `gatsby-plugin-netlify-cms`. Run the following command in the terminal at the root of your site:
+
+```sh
+npm install --save netlify-cms gatsby-plugin-netlify-cms
```
Comment (verythorough) [inline website/content/docs/using-netlify-cms-with-gatsby.md]: Nice catch on yarn vs. npm.

File: website/content/docs/using-netlify-cms-with-gatsby.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+### Push to GitHub
+
+It's now time to commit your changes and push to GitHub. The Gatsby starter initializes Git automatically for you, so you only need to do:
+
+```bash
+git add .
+git commit -m "Initial Commit"
+git remote add origin git@github.com:user/repo.git
```
Comment (verythorough) [inline website/content/docs/using-netlify-cms-with-gatsby.md]: I'd switch this to an https url, so folks without SSH keys won't get tripped up. For those who use SSH, we can be pretty safe assuming they'll know to translate the url.

I'd also recommend giving a bit more guidance in the url, like: 
`https://github.com/YOUR_USERNAME/NEW_REPO_NAME.git`

Comment (krismorf) [reply website/content/docs/using-netlify-cms-with-gatsby.md]: fixed 👌

**Comment by tomrutgers** [comment]
@verythorough @erquhart Right, menu placement! We currently have all the customization in the 'Guides' section (and a 'Contributor Guide' in the Contributing section to make it a bit more confusing), which leaves little room for the actual guides.. Would it make sense to reorganizing the navigation and have something like below? I'm aware that's out of the scope of this PR, but it might be worth including it in the original issue. #2092 

## Guides
- Gatsby
- Hugo
- Middleman
- ...

## Customization 
- Custom Widgets
- Custom Previews
- Custom Editor components

##Contributing
- How to contribute
- Architecture

**Comment by verythorough** [comment]
@tomrutgers that menu org sounds perfect. Also agree that it's out of scope here! 

For this PR, @krismorf, if you can add the front matter I described, I think we can merge this one. 

Then, maybe you'd like to make the PR for the menu re-grouping? I'm happy to go over what that would entail, if you're interested.

**Comment by krismorf** [comment]
@verythorough the guide has been adjusted 🚀 I will open a new PR for the menu org.

**Comment by verythorough** [comment]
Woot! Now with the front matter, we can see it in preview:
https://deploy-preview-2117--netlify-cms-www.netlify.com/docs/using-netlify-cms-with-gatsby/

LGTM 🎉

**Comment by verythorough** [comment]
Added an issue for the menu re-ordering: https://github.com/netlify/netlify-cms/issues/2149

**Comment by tomrutgers** [comment]
I don't know Gatsby all that much, but this looks really easy to follow! 

@verythorough It's been a while since I've seen a deploy preview posted by your account 😂

**Comment by tomrutgers** [comment]
Tried it out today, very easy to follow! Nice.
I agree with @Benaiah. I think we should point out that we do not follow the `content/blog-post/index.md` structure early on.

**Comment by krismorf** [comment]
@tomrutgers @Benaiah thanks for the feedback. I will mention that Netlify CMS does not follow the default folder structure. 

The guide works because in `gatsby-config.js` he have this:
```jsx
{
  resolve: `gatsby-source-filesystem`,
  options: {
    path: `${__dirname}/content/blog`,
    name: `blog`,
  },
}
```

and in `config.yml` we point to that directory:
```yml
collections:
    folder: "content/blog"
```

Should we mention this as well? @verythorough can you please help with the wording?

**Comment by erquhart** [comment]
@krismorf wording:

```markdown
**Note:** each existing blog post in `gatsby-starter-blog` is stored in its own directory. Netlify CMS doesn't work with files that are stored this way, so you won't be able to see or edit the starter's included sample blog posts in Netlify CMS, but you will be able to create and edit new posts.

A feature request for Netlify CMS to support entries stored in individual directories can be found here: [`netlify/netlify-cms#1472](https://github.com/netlify/netlify-cms/issues/1472).
```

Any further additions can happen after this is merged. Thanks so much for digging into this Kris!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by tomrutgers** [comment]
@Benaiah Can we get this merged?

**Comment by talves** [comment]
@krismorf thanks for working on this and your patience. We hope things can move quicker on something so easy to review!

**Comment by erquhart** [comment]
I think @tomrutgers just realized he's a maintainer lol

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**